### DISCUSSION TITLE
Custom Deep Feature Matching Node

### DISCUSSION THREAD

**Discussion by mirkosprojects** [alicevision/Meshroom#2130] [Q&A]
Hello there,

as part of a research into Deep Feature Matching Algorithms, I built a [custom node](https://github.com/mirkosprojects/MeshroomDFM) for Meshroom, that uses the [deep feature matching algorithm](https://github.com/ufukefe/DFM) from @ufukefe.
The node seems to generate matches, that I save in the same matches.txt file as the native Feature Matching node, however in most of my experiments, the pipeline fails at the Structure from Motion step without an explicit error message. The [log.txt](https://github.com/alicevision/Meshroom/files/12160560/log.txt) and [log1.txt](https://github.com/alicevision/Meshroom/files/12160616/log1.txt) suggest, that SfM can't find tracks, which is where I'm stuck at the moment. 

Some points, that might be worth mentioning:
- I use "dspsift" as the label of the feature extraction algorithm inside the matches.txt file, since I didn't know, if I could use a custom name. (Can be seen in the matches.txt example [here](https://github.com/mirkosprojects/MeshroomDFM#deepfeaturematching)).
- Every image pair has it's own matches.txt file, since they tend to get very large.
- The DFM-Algorithm does not return a descriptor, instead feature extraction and feature matching are merged into one algorithm, that only returns the x and y positions of matched features.

It would be nice if someone more knowledgeable on the development of meshroom nodes could help me with this problem. Thanks in advance!

**Comment by natowi** [comment]
I´d take a look at 

https://github.com/alicevision/AliceVision/blob/develop/src/software/pipeline/main_sfmTriangulation.cpp

https://github.com/alicevision/AliceVision/blob/develop/src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.hpp

https://github.com/alicevision/AliceVision/blob/develop/src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp

to understand what the process (Fuse matches into tracks) is doing.

**Comment by fabiencastan** [comment]
@mirkosprojects Interesting project! Sorry for the delay, I missed the message.
It should work, maybe just an index issue.
Would you be available to make a call to analyze your issue together?
You can contact me at: fabien.castan@technicolor.com

**Reply by mirkosprojects** [reply]
Thank you for your help and sorry for the delay.
Unfortunately I'm not currently working on this project and don't have access to the PC running it, as I've finished my studies at Hochschule Karlsruhe.
However I'm confident that this project will be picked up by other students in the future and they'd be happy to get your help!
